dashboard/app/controllers/home_controller.rb

```diff
@@ -115,6 +111,9 @@ def init_homepage
             linkToOverview: script_path(script),
             linkToLesson: script_next_path(script, 'next')
           }
+
+          # Don't include this course in the regular set of recent courses.
+          @recent_courses.reject! {|item| item[:id] == script[:name]}
```

Comment:
At last for a few more days (until we turn on student homepage), the `/courses` page depends on getting the full list from `recent_courses_and_scripts`.  

Even then (though I'm not familiar with the history of the primary course vs. the "rest" of courses), it feels like `recent_courses_and_scripts` on the User should probably return all recent courses and scripts, primary or otherwise.